clients/csharp-dotnet-core/voice-assistant-test/tool/BotConnector.cs

```diff
@@ -529,11 +545,9 @@ private void SpeechBotConnector_Recognized(object sender, SpeechRecognitionEvent
             if (e.Result.Reason == ResultReason.RecognizedSpeech)
             {
                 this.RecognizedText = e.Result.Text;
+                this.LengthOfSpeechInWavFile = (int)e.Result.Duration.TotalMilliseconds;

```

Comment:
What is this? does that give you the duration of the speech in the input stream? Is this documented somewhere?

Reply:
You read LengthOfSpeechInWavFile from the config file (from the WAVFile string), but you also get it here... something is wrong. 
Using e.Result.Duration.TotalMilliseconds may be the default value to use if the WAVFile string does NOT supply the "length of speech in WAV file" duration. But that depends on what e.Result.Duration really gives you. We can ask around if this is not documented or vaguely documented.

Reply:
e.Result.Duration is the length of recognized speech from Speech SDK

https://docs.microsoft.com/en-us/dotnet/api/microsoft.cognitiveservices.speech.recognitionresult.duration?view=azure-dotnet#Microsoft_CognitiveServices_Speech_RecognitionResult_Duration